dash/dash.py

```diff
@@ -1565,6 +1566,28 @@ def get_asset_url(self, path):
 
         return asset
 
+    def get_relative_path(self, path):
```

Comment:
Naming options:
- `dcc.Link(href=app.get_relative_path('/page-2'))`
- `dcc.Link(href=app.get_relative_url('/page-2'))`
- `dcc.Link(href=app.get_rel_path('/page-2'))`
- `dcc.Link(href=app.get_rel_url('/page-2'))`
- `dcc.Link(href=app.relative_path('/page-2'))`
- `dcc.Link(href=app.relative_url('/page-2'))`
- `dcc.Link(href=app.rel_path('/page-2'))`
- `dcc.Link(href=app.rel_url('/page-2'))`
- `dcc.Link(href=app.relpath('/page-2'))`
- `dcc.Link(href=app.relurl('/page-2'))`

Notes:
- It is technically a `path` and not a full URL although folks might think that "path" is corresponding to the file system path.
- We use `get_` in other places like `get_asset_url`. Do we keep it for consistency? I'd prefer to move `get_asset_url` usage over to this function, so folks just need to learn e.g. `app.relpath('/assets/logo.png')` and `app.relpath('/page-2')`

I think I prefer `app.relative_path('/page-2')`.

Reply:
`get_asset_url` is still nice in order to decouple the usage from `assets_url_path`. Also in that case it *is* a file path (that you pass in - though it occurs to me it's a non-Windows path, we may want to move or copy the `.replace("\\", "/")` that we currently have in `_on_assets_change` into `get_asset_url` if we're expecting windows users to directly use `get_asset_url`) - so it seems like our nomenclature is distinguishing file path from (partial) url, rather than url path from full url.

The other thing I'm wondering is whether we need the reverse operation, stripping off `requests_pathname_prefix`, for use inside `dcc.Location` callbacks? If you think that would be useful, then we should name these two methods for clarity in tandem.

Reply:
For use in `dcc.Location`, I'm now thinking that we could include this function in the callback url matching logic like:
```python
@app.callback(...)
def display_content(path):
    if (path is None or 
            app.get_relative_path(path) == app.get_relative_path('/')):
        return chapters.home
    elif app.get_relative_path(path) == app.get_relative_path('/page-1')
        return chapters.page_1
    elif app.get_relative_path(path) == app.get_relative_path('/page-2')
        return chapters.page_2
```

Or, rather, the trailing slashes case:
```python
@app.callback(...)
def display_content(path):
    if path is None or app.get_relative_path(path) == app.get_relative_path("/"):
        return chapters.home
    elif app.get_relative_path(path) in [
        app.get_relative_path("/page-1"),
        app.get_relative_path("/page-1/"),
    ]:
        return chapters.page_1
    elif app.get_relative_path(path) in [
        app.get_relative_path("/page-2"),
        app.get_relative_path("/page-2/"),
    ]:
        return chapters.page_2
```

Reply:
That last example actually exposes another missing case, `''`, fixed in 868e93a

The documented example should actually be
```python
@app.callback(...)
def display_content(path):
    if path is None or app.get_relative_path(path) in [
        app.get_relative_path("/"),
        app.get_relative_path(""),
    ]:
        return chapters.home
    elif app.get_relative_path(path) in [
        app.get_relative_path("/page-1"),
        app.get_relative_path("/page-1/"),
    ]:
        return chapters.page_1
    elif app.get_relative_path(path) in [
        app.get_relative_path("/page-2"),
        app.get_relative_path("/page-2/"),
    ]:
        return chapters.page_2
```

Reply:
I suppose, but folks may appreciate simplifying it (including stripping the trailing `/` and maybe even the leading one, if we were to have `get_relative_path` work add a leading `/`):
```py
@app.callback(...)
def display_content(path):
    page_name = app.trim_path(path)
    if not page_name:  # None or ''
        return chapters.home
    elif page_name == 'page-1':
        return chapters.page_1
    if page_name == "page-2":
        return chapters.page_2
```
At which point you could even avoid having to reference all the pages individually, since `page_name` would be fully normalized:
```py
@app.callback(...)
def display_content(path):
    page_name = app.trim_path(path)
    if not page_name:  # None or ''
        return chapters.home
    return getattr(chapters, page_name.replace("-", "_"))
    # or make chapters be a dict that maps without the replacement
```

Reply:
Yeah, that's way better! I'll work on a `trim_path` method. There should definitely be symmetry between the naming of `trim_path` and `relative_path`, whatever we decide on.

Reply:
Alright, I added `strip_relative_path` which would match the naming of `get_relative_path` in 20dd95f